backend-peer/src/lib/services/cloudagent.ts

```diff
@@ -72,7 +81,39 @@ export class CloudagentManager {
     return agentInfo as AgentInfo
   }
 
-  receiveImplicitInvitation = async (did: string) => {
+  receiveImplicitInvitation = async (
+    did: string,
+    waitUntilCompleted: boolean = false
+  ) => {
+    let connectionId: string | undefined = undefined
+    let connectionRecordPromise: Promise<ConnectionRecord> | undefined =
+      undefined
+    if (waitUntilCompleted) {
+      connectionRecordPromise = new Promise<ConnectionRecord>(
+        (resolve, reject) => {
+          const timeout = setTimeout(() => {
+            reject(new InternalError(`Could not create a connection`))
+          }, this.env.CONNECTION_REQUEST_TIMEOUT_MS)
+          const connectionEventListener = (
+            connectionRecord: ConnectionRecord
+          ) => {
+            if (
+              connectionRecord.id === connectionId &&
+              connectionRecord.state === 'completed'
+            ) {
+              clearTimeout(timeout)
+              this.eventEmitter.off(
+                EventType.Connection,
+                connectionEventListener
+              )
+              return resolve(connectionRecord)
+            }
+          }
+          this.eventEmitter.on(EventType.Connection, connectionEventListener)
+        }
+      )
```

Comment:
could this all go in the if block on L144? Then wouldn't need the `let` variables

Reply:
@jonmattgray  I think the point there is that wait until completed if block is normally skipped and we don't want that to be the case always ? e.g  in the endpoints that you made for creating the connections we don't have to wait for the connection to reach a completed stage, right?